app/__main__.py

```diff
@@ -50,11 +54,15 @@ def main(computation_manager: ComputationManager):
                         active_mode = False
                 elif not subproblem.is_alive():
                     identifier = subproblem.identifier
-                    result = subproblem.result
-                    computation_manager.handle_completed(subproblem)
-                    broadcast_result(subproblem, broker)
-                    subproblem = None
-                    logger.info(f'Subproblem #{identifier} has ended (result: {result}).')
+                    if computation_manager.pool.current_subproblem_id is None:
+                        subproblem = None
+                        logger.info(f'Subproblem #{identifier} has been dropped.')
+                    else:
+                        result = subproblem.result
+                        computation_manager.handle_completed(subproblem)
+                        broadcast_result(subproblem, broker)
+                        subproblem = None
+                        logger.info(f'Subproblem #{identifier} has ended (result: {result}).')
```

Comment:
`subproblem = None` is executed regardless of the condition. Thus, you may place it after the if-else block.